org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/converter/ZigBeeBaseChannelConverter.java

```diff
@@ -151,6 +151,12 @@
      */
     protected int pollingPeriod = POLLING_PERIOD_DEFAULT;
 
+    /**
+     * The smallest of all maximum reporting periods configured for the attributes used by the converter. By default it
+     * is set to {@code Integer.MAX_VALUE}, because we assume that not every converter implements reporting.
+     */
+    protected int minimalMaximumReportingPeriod = Integer.MAX_VALUE;
```

Comment:
This may need to be aligned with the reporting configuration?
Also, as per other comments - I don't really think this is a very descriptive name.

Reply:
Yes, it IS aligned with the reporting configurations, please see the changes I made to the converters.

Reply:
Ok, maybe I missed this, but I didn't see any reference to the ```ZclReportingConfig``` class. My apologies - I will take a better look later.

Reply:
Maybe we have a misunderstanding here since I am not yet aware of your planned reporting configurations, see my comment https://github.com/openhab/org.openhab.binding.zigbee/pull/412#issuecomment-473198733 regarding this.

What I do in the converter is, I call the new method

```
handleReportingResponse(CommandResult reportResponse, int reportingFailedPollingInterval,
            int reportingSuccessMaxReportInterval)
```

which sets the `pollingPeriod` and `minimalMaximumReportingPeriod` fields accordingly, depending on whether the reporting configuration request was successful or not.

Reply:
> What I do in the converter is, I call the new method

Yes, that I saw. My question is if this isn't the appropriate time to start using the new class rather than have to change again. This function was added a month or two back, but not rolled into the converters yet.

Reply:
Which function was added a month or two back?

The

```
handleReportingResponse(CommandResult reportResponse, int reportingFailedPollingInterval,
            int reportingSuccessMaxReportInterval)
```

is my function which I added in this PR :)

Reply:
As above - it is the ```ZclReportingConfig``` class which is designed to allow users to configure all of these timers that are currently hard coded here.

I think (but might be wrong as I wrote this a month or two back) that it may just be a case of changing to use methods in this class to get the times, rather than hard coding them.

Reply:
Ah I see, so what you mean is that the converters should make use of the `ZclReportingConfig` instead of using hard coded values.

Since this is not the case for now, I have used these hardcoded values as input to my `handleReportingResponse` method. If the `ZclReportingConfig` gets used in the future, one can easily replace them.

I think this is out of scope of this PR and would only blow up the lines of code. I think a separate should take care of making the channel converters use the `ZclReportingConfig`, WDYT?

Reply:
@cdjackson may I get back to you on this issue?

I see a usage of the `ZclReportingConfig` in 

https://github.com/openhab/org.openhab.binding.zigbee/blob/0c7dbd66a6a7df79ce9e4c89e1b423060a6e8c14/org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterSwitchOnoff.java#L119-L120

for example.

However, when the device is configured, this configuration is not used, but the hard-coded default values are used.

https://github.com/openhab/org.openhab.binding.zigbee/blob/0c7dbd66a6a7df79ce9e4c89e1b423060a6e8c14/org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterSwitchOnoff.java#L68-L72

So from my point of view the change that I made does not touch the implementation you did with the `ZclReportingConfig` at all. it is totally independent from it.

Once the created `ZclReportingConfig` in a `initializeConverter()` is used in the `initializeDevice()`  the changes I make in this PR can flow into it.

Making use of the configuration set in the `initializeConverter()` inside the `initializeDevice()` should IMHO be done in a separate PR as it hasn't been done so far for any converter.

So how shall we proceed on this PR?